The docker image job is broken on master

The `docker_push_image` CI job fails on master with

```
$ Run docker tag nix:$NIX_VERSION nixos/nix:$NIX_VERSION
  docker tag nix:$NIX_VERSION nixos/nix:$NIX_VERSION
  shell: /usr/bin/bash -e {0}
  env:
    CACHIX_NAME: nixos-nix-install-tests
    NIX_VERSION: 
    CACHIX_SIGNING_KEY: ***
Error parsing reference: "nix:" is not a valid repository/tag: invalid reference format
Error: Process completed with exit code 1.
```

(see https://github.com/NixOS/nix/runs/6797770409?check_suite_focus=true for example)

This started roughly at the same time as the nixpkgs bump, so I guess it must be related

cc @garbas @cole-h

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -88,7 +88,7 @@ jobs:
         fetch-depth: 0
     - uses: cachix/install-nix-action@v17
     - run: echo CACHIX_NAME="$(echo $GITHUB_REPOSITORY-install-tests | tr "[A-Z]/" "[a-z]-")" >> $GITHUB_ENV
-    - run: echo NIX_VERSION="$(nix-instantiate --eval -E '(import ./default.nix).defaultPackage.${builtins.currentSystem}.version' | tr -d \")" >> $GITHUB_ENV
+    - run: echo NIX_VERSION="$(nix --experimental-features 'nix-command flakes' eval .\#default.version | tr -d \")" >> $GITHUB_ENV
     - uses: cachix/cachix-action@v10
       if: needs.check_cachix.outputs.secret == 'true'
       with:

